### PR TITLE
Issue45 FAQ Block : Accordian like faq block

### LINKED ISSUE #45: Block:FAQ
Seen on https://www.caesars.com/caesars-palace/things-to-do/garden-of-the-gods-pool-oasis

image

### PR DESCRIPTION
**Pull request by jfait**
Fix #45 

## 🔗 Test URLs:
  
Before: https://main--caesars--hlxsites.hlx.page/caesars-palace/drafts/jfait/faq
After: https://issue-45-faq-block--caesars--hlxsites.hlx.page/caesars-palace/drafts/jfait/faq

Live Example:  https://www.caesars.com/caesars-palace/things-to-do/garden-of-the-gods-pool-oasis#894cd7a7-8e63-3d2f-9c22-236cb2dcda8e

## 📝 Description:
  
FAQ block with css and js (decoration and interactivity to expand and collapse)

### CODE REVIEW

File: caesars-palace/blocks/faq/faq.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    Array.from(row.children)
+      .forEach((childElement) => {
+        questionContentText.appendChild(childElement);
+      });
+
+    row.appendChild(questionContentText);
+
+    const expandIcon = document.createElement('div');
+    expandIcon.innerHTML = '<div class="toggle" role="button" aria-pressed="false"><span data-testid="toggle-span"><i data-testid="toggle-icon"></i></span></div>';
```
Comment (chrismborland) [inline caesars-palace/blocks/faq/faq.js:38]: For my own knowledge, is it better to create the elements via `innerHTML` or `document.createElement(<element-name>)`?

File: caesars-palace/blocks/faq/faq.css [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+/* Header/Title */
+div.faq-container div > h1 {
+    width: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin: 0 auto;
+    margin-bottom: 24px;
```
Comment (tmathern) [inline caesars-palace/blocks/faq/faq.css:29]: I think there is a variable for that value

File: caesars-palace/blocks/faq/faq.css [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    color: var(--static-color-white);
+    font-size: 36px
+}
+
+/* FAQ Row */
+div.faq-container .questions-content {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    padding: 32px;
```
Comment (tmathern) [inline caesars-palace/blocks/faq/faq.css:39]: I think there is a variable for that value

File: caesars-palace/blocks/faq/faq.css [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+div.faq-container .questions-content:last-of-type {
+    border-bottom: 1px solid rgb(255 255 255 / 40%);
+    margin-bottom: 64px
+}
+
+div.faq-container .questions-content-text {
+    flex: 1;
+    flex-direction: column;
+    margin-right: 16px;
```
Comment (tmathern) [inline caesars-palace/blocks/faq/faq.css:54]: I think there is a variable for that value

File: caesars-palace/blocks/faq/faq.css [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    padding: 32px;
+    border-top: 1px solid rgb(255 255 255 / 40%);
+    border-bottom: 1px solid transparent;
+    cursor: pointer;
+    transition: border-bottom-color .15s ease-in-out, border-top-color .15s ease-in-out
+}
+
+div.faq-container .questions-content:last-of-type {
+    border-bottom: 1px solid rgb(255 255 255 / 40%);
+    margin-bottom: 64px
```
Comment (tmathern) [inline caesars-palace/blocks/faq/faq.css:48]: I think there is a variable for that value

File: caesars-palace/blocks/faq/faq.css [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+div.faq-container .questions-content-text > div:nth-child(1) {
+    font-weight: bold;
+    transition: color .15s ease-in-out
+}
+
+/* Answer Cell */
+div.faq-container .questions-content-text > div:nth-child(2) {
+    display: none;
+    visibility: hidden;
+    margin-top: 16px;
```
Comment (tmathern) [inline caesars-palace/blocks/faq/faq.css:68]: I think there is a variable for that value

**Review by tmathern — APPROVED** [review]

**Comment by jfait** [comment]
Updated linting and vars.
Currently allows for any number of answers to be open (site forces only 1).
Will log issue to revisit the only1

**Pull request merged**